File: src/php/Whetstone/WhetstoneBenchmark.php
<?php

namespace EntityFX\NetBenchmark\Core\Whetstone {

    use EntityFX\NetBenchmark\Core\Writer;
    use EntityFX\NetBenchmark\Core\Generic\BenchmarkBase;

    class WhetstoneBenchmark extends BenchmarkBase {

        private $dhrystone;

        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Ratio = 20;
            $this->dhrystone = new Whetstone($printToConsole);
        }

        public function Warmup($aspect = 0.05) {
        }

        protected function BenchImplementation() {
            return $this->dhrystone->bench(false);
        }

        protected function PopulateResult($benchResult, $result) {
            $benchResult["Points"] = $result["MWIPS"] * $this->Ratio;
            $benchResult["Result"] = $result["MWIPS"];
            $benchResult["Units"] = "MWIPS";
            $benchResult["Output"] = $result["Output"];
            return $benchResult;
        }
    }
}

File: src/lua/whetstone/whetstoneBenchmark.lua
require "whetstone/whetstone"

WhetstoneBenchmark = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.name = "WhetstoneBenchmark"
    a.ratio = 1
    a.whetstone = Whetstone(printToConsole)
end)

function WhetstoneBenchmark:benchImplementation()
    return self.whetstone:bench()
end

function WhetstoneBenchmark:warmup(aspect)

end

function WhetstoneBenchmark:populateResult(benchResult, result)
    -- dumpTable(result, 200)
    benchResult.points = result.mwips * self.ratio
    benchResult.result = result.mwips
    benchResult.units = "MWIPS"
    benchResult.output = result.output
    return benchResult
end